src-electron/zcl/zcl-loader-silabs.js

```diff
@@ -2303,65 +2327,58 @@ function parseFeatureConformance(operand) {
     // check '$' key is not the only key in the object to handle special cases
     // e.g. '<optionalConform choice="a" more="true"/>'
     if (insideTerm && Object.keys(insideTerm).toString() != '$') {
-      return `[${parseFeatureConformance(operand.optionalConform[0])}]`
+      return `[${parseConformanceRecursively(operand.optionalConform[0])}]`
     } else {
       return 'O'
     }
+  } else if (operand.otherwiseConform) {
+    return Object.entries(operand.otherwiseConform[0])
+      .map(([key, value]) => parseConformanceRecursively({ [key]: value }))
+      .join(', ')
+  } else if (operand.notTerm) {
+    // need to surround terms inside a notTerm with '()' if it contains multiple terms
+    // e.g. !(A | B) or !(A & B)
+    // able to process multiple parallel notTerms, e.g. !A & !B
+    return operand.notTerm
+      .map((term) => {
+        let nt = parseConformanceRecursively(term)
+        return nt.includes('&') || nt.includes('|') ? `!(${nt})` : `!${nt}`
+      })
+      .join(` ${parentJoinChar} `)
+  } else if (operand.andTerm || operand.orTerm) {
+    // process andTerm and orTerm in the same logic
+    // when joining multiple orTerms inside andTerms, we need to
+    // surround them with '()', vice versa for andTerms inside orTerms
+    // e.g. A & (B | C) or A | (B & C)
```

Comment:
You should use antlr for these kid of things in the future.

Reply:
Yes we could switch to ANTLR if conformance parsing became too complicated to be parsed one day